dev-tools/generate_notice

```diff
@@ -0,0 +1,4 @@
+go list -m -json all | go run go.elastic.co/go-licence-detector \
+    -noticeTemplate dev-tools/NOTICE.txt.tmpl \
+    -noticeOut NOTICE.txt \
+    -depsOut ""
```

Comment:
is this a shell script? Maybe let's add `#!` then.

Reply:
Added.